controllers/dockermachine_controller.go

```diff
@@ -329,3 +333,40 @@ func (r *DockerMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		).
 		Complete(r)
 }
+
+func (r *DockerMachineReconciler) delete(
+	ctx context.Context,
+	cluster *capiv1alpha2.Cluster,
+	machine *capiv1alpha2.Machine,
+	dockerMachine *infrastructurev1alpha2.DockerMachine,
+) (ctrl.Result, error) {
+
+	if isControllerPlaneMachine(machine) {
+		err := actions.DeleteControlPlane(cluster.GetName(), machine.GetName())
+		if err != nil {
+			r.Log.Error(err, "Error deleting control plane node", "nodeName", machine.GetName())
+			return ctrl.Result{RequeueAfter: time.Second * 30}, err
+		}
+	} else {
+		err := actions.DeleteWorker(cluster.GetName(), machine.GetName())
+		if err != nil {
+			r.Log.Error(err, "Error deleting worker node", "nodeName", machine.GetName())
+			return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+		}
+	}
+	err := r.Client.Delete(ctx, dockerMachine)
```

Comment:
This will be a noop, instead remove the finalizer and patch the machine, k8s will cleanup the resource afterwards.